compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -495,6 +496,25 @@ pub(in crate::rmeta) fn provide(providers: &mut Providers) {
             tcx.untracked().cstore.leak();
             tcx.arena.alloc_from_iter(CStore::from_tcx(tcx).iter_crate_data().map(|(cnum, _)| cnum))
         },
+        doc_masked_crates: |tcx, LocalCrate| {
```

Comment:
Can you put this functionality in a function and only call that function in the closure here, please? I think keeping the `provide` functions short is preferable and is what seems to be done in most places in the compiler.